### PR TITLE
[#176392795] Add pr template

### PR DESCRIPTION
**Pull request by balanza**
#### Description
Add a Pull Request template ton this repository

#### Motivation and Context
We produce a large number of PR with poor or no description at all. We may benefit a template to enforce the best practice of having an exhaustive description.

#### How Has This Been Tested?
This same PR

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Kindly inspired by: https://github.com/stevemao/github-issue-templates/blob/master/questions-answers/PULL_REQUEST_TEMPLATE.md

### CODE REVIEW

**Review by gunzip — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#### Description
+<!--- Describe your changes in detail -->
+
+#### Motivation and Context
+<!--- Why is this change required? What problem does it solve? -->
+
+#### How Has This Been Tested?
+<!--- Please describe in detail how you tested your changes. -->
+<!--- Include details of your testing environment, tests ran to see how -->
+<!--- your change affects other areas of the code, etc. -->
```
Comment (AleDore) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: This section means that we should update PR's title once we have tested our changes. Normally we open PR's just with the Description and the context. I was wondering if the test docs can be moved away from PR's description.

**Review by AleDore — COMMENTED** [review]
LGTM apart from a little doubt :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!--- Please always add a PR description as if nobody knows anything about the context these changes come from. -->
+<!--- Even if we are all from our internal team, we may not be on the same page. -->
+<!--- Write this PR as you were contributing to a public OSS project, where nobody knows you and you have to earn their trust. -->
+<!--- This will improve our projects in the long run! Thanks. -->
+
+#### Description
+<!--- Describe your changes in detail -->
+
+#### Motivation and Context
+<!--- Why is this change required? What problem does it solve? -->
```
Comment (gunzip) [inline .github/PULL_REQUEST_TEMPLATE.md]: Isn't decription the same as motivation and context? (it seems reduntant)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (balanza) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: Ok maybe it's too much writing for the intended effort.
What I intended here was "just drop a line of how did you test this". That could be:
* execute http request in local environment/io-mock
* I wrote extensive tests
* nothing

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (balanza) [reply .github/PULL_REQUEST_TEMPLATE.md]: I wanted to lead the writer in be explicit in both motivation and context, as I see most PRs merely summarize _what_ has been done and they give the _why_ for granted.

Maybe we can collapse both paragraph as one: `#### Description, motivation and context`?

Comment (gunzip) [reply .github/PULL_REQUEST_TEMPLATE.md]: maybe we can be more explicit with titles (ie. "List of changes", "Motivation and context")

Comment (balanza) [reply .github/PULL_REQUEST_TEMPLATE.md]: So the suggestion is to keep those separated, but to change `Description` with `List of Changes`?

Comment (gunzip) [reply .github/PULL_REQUEST_TEMPLATE.md]: yes

**Review by gunzip — APPROVED** [review]

**Pull request merged**